src/back/spv/test_framework.rs

```diff
@@ -0,0 +1,70 @@
+use crate::back::spv::Instruction;
+use spirv::{Op, Word};
+
+pub(crate) struct SpecRequirements {
+    op: Op,
+    wc: u32,
+    type_id: bool,
+    result_id: bool,
+    operands: bool,
+}
+
+impl SpecRequirements {
+    pub fn new(op: Op, wc: u32, type_id: bool, result_id: bool, operands: bool) -> Self {
+        SpecRequirements {
+            op,
+            wc,
+            type_id,
+            result_id,
+            operands,
+        }
+    }
+}
+
+/// Basic validation for checking if the instruction complies to the spec requirements
+pub(crate) fn validate_spec_requirements(
+    requirements: SpecRequirements,
+    instruction: &Instruction,
+) {
+    assert_eq!(requirements.op, instruction.op);
+
+    // Pass the assert if the minimum referred wc in the spec is met
+    assert!(instruction.wc >= requirements.wc);
```

Comment:
I am now not checking variable word count. Is this something we also want to support now, or would the minimum word count be suffice for now?